added route change announcement

JanAckermann opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for Web. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" in case the PR still has open tasks
- Set label "backport-request" if backport is needed
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
Enhancement: Implement live region updates on route changes

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes https://github.com/owncloud/web/issues/4346

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/26169327/111342350-aa866a80-867a-11eb-90dd-6165f57cc661.png)


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] ...

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/web?pullRequest=4812) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/web?pullRequest=4812) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: JanAckermann<br/>:x: JammingBen<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/web?pullRequest=4812) it.</sub>

JanAckermann commented:
look at div announcer :rocket:

JanAckermann commented:
@pascalwengerter @kulmann Could you have a closer look?
* I removed vue-announcer and replaced it with the oc-hidden-announcer
* Needed to add i18n to the web-runtime router

JanAckermann commented:
Dunno whats up with license/cla

JanAckermann commented:
@kulmann i18n interpolate introduced

kulmann commented on packages/web-runtime/src/App.vue:
```diff
@@ -321,11 +324,19 @@ export default {
+    announceRouteChange(route) {
+      const pageTitle = this.extractPageTitleFromRoute(route, false)
+
+      const translated = this.$gettext(' Navigated to %{ pageTitle }')
```
Why the blank before `Navigate`?

kulmann commented on packages/web-runtime/src/App.vue:
```diff
@@ -83,7 +84,7 @@
 <script>
 import 'inert-polyfill'
-import { mapGetters, mapState, mapActions } from 'vuex'
+import {mapActions, mapGetters, mapState} from 'vuex'
```
```suggestion
import { mapActions, mapGetters, mapState } from 'vuex'
```
... to make the linter happy ;-)

kulmann commented on packages/web-runtime/src/App.vue:
```diff
@@ -321,11 +324,18 @@ export default {
-        titleSegments.unshift(route.meta.pageTitle)
+    announceRouteChange(route) {
+      const pageTitle = this.extractPageTitleFromRoute(route, false)
+      const translated = this.$gettext(' Navigated to %{ pageTitle }')
```
```suggestion
      const translated = this.$gettext('Navigated to %{ pageTitle }')
```
sorry, I meant this blank ;-)

kulmann reviewed (approved):
💪

kulmann commented:
Added @marcus-herrmann as reviewer, since I lack experience with the live region updates and cannot tell if this is sufficient. Code looks good. Could you take a look @marcus-herrmann ?

marcus-herrmann reviewed (approved):
LGTM. But is it possible to see this working live soon?

kulmann commented:
> LGTM. But is it possible to see this working live soon?

As soon as it's merged it will be grabbed by the next auto-reset on web.owncloud.com. Happens at least once per day afaik. So if we can merge it today it would be visible by tomorrow.

marcus-herrmann commented:
Ok. So web.owncloud.com is newer than ocis-a11y.owncloud.com then. Interestingly I see a different side wide search function there (compared to ocis-a11y where I audited). Also, the files table fix doesn't seem to be on "web."

kulmann commented:
> Ok. So web.owncloud.com is newer than ocis-a11y.owncloud.com then. Interestingly I see a different side wide search function there (compared to ocis-a11y where I audited). Also, the files table fix doesn't seem to be on "web."

ocis doesn't have search, that's why it's disabled there. Files table is still not merged. ;-) The PR is close to being merged though.

marcus-herrmann commented:
> ocis doesn't have search, that's why it's disabled there. Files table is still not merged. ;-) The PR is close to being merged though.

Oh I see. But if ocis-a11y is not representative (at least in the regards of search), it is not ideal as a basis for an accessibility audit?

marcus-herrmann commented:
> Oh I see. But if ocis-a11y is not representative (at least in the regards of search), it is not ideal as a basis for an accessibility audit?

Because I see a WCAG 3.3.2 violation there: https://www.w3.org/TR/WCAG20-TECHS/H32.html (sorry for hijacking the ticket)

pascalwengerter reviewed (approved):
LGTM! 🔥

pascalwengerter commented:
Works well for me, locally. @marcus-hermann can this be merged and is your concerns regarding a possible `WCAG 3.3.2 violation` (I guess regarding the search submit, but not sure if related to route change announcements?) tracked somewhere already?

kulmann commented:
Merging, as the comment about search was not related to the announcer. Question about whether or not it is tracked somewhere is still valid though. 😅

marcus-herrmann commented:
> Works well for me, locally. @marcus-hermann can this be merged and is your concerns regarding a possible WCAG 3.3.2 violation (I guess regarding the search submit, but not sure if related to route change announcements?) tracked somewhere already?

@pascalwengerter: Sorry, this is unrelated to this ticket, so it would be absurd  to have been a merge blocker. I will create a new issue, though, since this could/will be critical for WCAG/BITV but was not part of the version I audited.

pascalwengerter commented:
> > Works well for me, locally. @marcus-hermann can this be merged and is your concerns regarding a possible WCAG 3.3.2 violation (I guess regarding the search submit, but not sure if related to route change announcements?) tracked somewhere already?
> 
> @pascalwengerter: Sorry, this is unrelated to this ticket, so it would be absurd to have been a merge blocker. I will create a new issue, though, since this could/will be critical for WCAG/BITV but was not part of the version I audited.

Was more of a rethoric question ;) thank you & looking forward to the new ticket (and more information on what's to be fixed)!

marcus-herrmann commented:
> Was more of a rethoric question ;) thank you & looking forward to the new ticket (and more information on what's to be fixed)!

Regarding that: https://github.com/owncloud/web/issues/4867
